File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    bool checkv(char s){
        if(s == 'a' || s== 'e' || s== 'i' || s== 'o' || s== 'u'){
            return true;
        }
        return false;
    }
    int maxVowels(string s, int k) {
        int cnt = 0;
        for(int i=0; i<k; i++){
            if(checkv(s[i])){
                cnt++;
            }
        }
        int maxcount = cnt;
        for(int i=k; i<s.length(); i++){
            if(checkv(s[i-k])){
                cnt--;
            }
            if(checkv(s[i])){
                cnt++;
            }
            maxcount = max(maxcount, cnt);
        }
        return maxcount;
    }
};

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def checkvo(self, s):
        if s == 'a' or s == 'e'  or s == 'i' or s == 'o' or s == 'u':
            return True
        return False

    def maxVowels(self, s: str, k: int) -> int:
        cnt = 0
        for i in range(k):
            if self.checkvo(s[i]):
                cnt += 1
        maxcnt = cnt
        for i in range(k, len(s)):
            if self.checkvo(s[i - k]):
                cnt -= 1
            if self.checkvo(s[i]):
                cnt += 1
            maxcnt = max(maxcnt, cnt)
        return maxcnt
            


        